pyccel/codegen/printing/pycode.py

```diff
@@ -660,15 +666,20 @@ def _print_For(self, expr):
     def _print_FunctionalFor(self, expr):
         body, iterators = self._find_functional_expr_and_iterables(expr)
         lhs = self._print(expr.lhs)
-        body = self._print(body.rhs)
-        for_loops = ' '.join(['for {} in {}'.format(self._print(idx), self._print(iters))
-                        for idx, iters in zip(expr.indices, iterators)])
+        condition = ''
+        if isinstance(body, Assign):
+            body = self._print(body.rhs)
+        else:
+            assert isinstance(body, ListAppend)
+            body = self._print(body.args[0])
 
-        name = self._aliases.get(type(expr),'array')
-        if name == 'array':
-            self.add_import(Import('numpy', [AsName(NumpyArray, 'array')]))
+        for_loops = ' '.join(f'for {self._print(idx)} in {self._print(iters)}{" if " + self._print(condition.blocks[0].condition) if condition else ""}'
+                             for idx, iters, condition in zip(expr.indices, iterators, expr.conditions))
 
-        return '{} = {}([{} {}])\n'.format(lhs, name, body, for_loops)
+        if isinstance(expr.class_type, NumpyNDArrayType):
+            self.add_import(Import('numpy', [AsName(NumpyArray, 'array')]))
+            return f'{lhs} = array([{body} {for_loops} {condition}])\n'
```

Comment:
Can `_get_numpy_name` be used here to ensure the expected name?
```suggestion
        if isinstance(expr.class_type, NumpyNDArrayType):
            array = self._get_numpy_name(NumpyArray)
            return f'{lhs} = array([{body} {for_loops} {condition}])\n'
```

Reply:
fixed